### ISSUE TITLE
`deno vendor` appears to not always respect type assertions in dynamic imports

### ISSUE THREAD

**Issue by cscheid** [denoland/deno#15634]
Somewhere between `1.22.0` and `1.25.0`, `deno vendor` stopped working for us over at https://github.com/quarto-dev/quarto-cli:

```
$ git clone git@github.com:quarto-dev/quarto-cli.git
Cloning into 'quarto-cli'...
$ cd quarto-cli/src
$ rm -rf vendor
$ deno vendor quarto.ts ../tests/test-deps.ts --importmap=./import_map.json
error: Expected ';', '}' or <eof> at file:///private/tmp/quarto-cli/src/resources/editor/tools/yaml/tree-sitter-yaml.json:1:8
```

This error appears in a few different places where we use a dynamic import of type `json` (we have to do it in this way because reasons, if you're curious):

- https://github.com/quarto-dev/quarto-cli/blob/main/src/core/lib/yaml-intelligence/parsing.ts#L57-L60
- https://github.com/quarto-dev/quarto-cli/blob/main/src/core/lib/yaml-intelligence/yaml-intelligence.ts#L1169-L1172
- https://github.com/quarto-dev/quarto-cli/blob/main/src/core/schema/utils.ts#L20-L27

Unfortunately, my attempts at a single-file minimal reproduction failed; I apologize.

**Comment by sylc** [comment]
deno vendor does not seem to support type assertion, even on non-dynamic import.  see the below basic repro

// version.json
```json
{
  "version": "0.0.1"
}
```

```ts
// main.ts
import version from "./version.json" assert { type: "json" };
console.log(version);

import * as datetime from "https://deno.land/std@0.140.0/datetime/mod.ts";
console.log(datetime.format(new Date(), "yyyy-MM-dd"));
```
output
```
$ deno vendor ./main.ts
error: Expected ';', '}' or <eof> at file:///C:/test/version.json:2:12
```

```
$ deno --version
deno 1.25.4 (release, x86_64-pc-windows-msvc)
v8 10.7.193.3
typescript 4.7.4
```

**Comment by dsherret** [comment]
I believe fixed via #16059
